ArrayList/ArrayList.cs

```diff
@@ -1,8 +1,580 @@
 ﻿using System;
+using System.Text;
 
-namespace ArrayList
+namespace List
 {
-    public class ArrayList
+    public class ArrayList<T> where T : IComparable<T>
     {
+        public int Length
+        {
+            get
+            {
+                return _length;
+            }
+            private set
+            {
+                if (value >= 0)
+                {
+                    _length = value;
+                }
+                else if (value < 0)
```

Comment:
Looks a bit strange. Could you explain when you expect this condition?

Reply:
In worst case you can use ternary

Reply:
If something goes wrong when reducing the length and there is an unnecessary operation.

Reply:
Replace with ternary

Reply:
ok